### PR TITLE
[docs] K8s docs intro polish and KubeRay architecture diagram

### PR DESCRIPTION
**Pull request by pcmoritz**
## Why are these changes needed?

This PR adds a KubeRay architecture diagram and an overview of what can be found in the Ray on Kubernetes documentation.

Thanks @scottsun94 for helping to design the graphics!

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md [hunk 1]
```diff
@@ -2,12 +2,32 @@
...
 ## Overview
 
-You can execute your distributed Ray programs on a Kubernetes cluster.
+In this section we cover how to execute your distributed Ray programs on a Kubernetes cluster.
 
-The [KubeRay Operator](https://ray-project.github.io/kuberay/components/operator/) provides a Kubernetes-native
-interface for managing Ray clusters. Each Ray cluster consist of a head pod and collection of worker pods.
-Optional autoscaling support allows the KubeRay Operator to size your Ray clusters according to the requirements
-of your Ray workload, adding and removing Ray pods as needed.
+The [KubeRay Operator](https://ray-project.github.io/kuberay/components/operator/) is the recommended way to do so. It provides a Kubernetes-native way to manage Ray clusters. Each Ray cluster consist of
```
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md:7]: nits:
`Using the Kuberay Operator is...`
`The KubeRay operator provides...`

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md [hunk 2]
```diff
@@ -2,12 +2,32 @@
...
+
+..
+  https://docs.google.com/drawings/d/1E3FQgWWLuj8y2zPdKXjoWKrfwgYXw6RV_FWRwK8dVlg/edit
+```
+
+
+Concretely you will learn how to:
+
+- Set up and configure Ray on an existing Kubernetes cluster or a managed Kubernetes service
+- Deploy batch Ray applications on Kubernetes via Ray Jobs
```
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md:25]: Does this RayJob controller? That functionality is fairly recently merged and experimental

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md [hunk 3]
```diff
@@ -2,12 +2,32 @@
...
+- Utilize GPUs to run Ray machine learning workloads (e.g. with Ray AIR)
+- Set up networking to connect your Ray applications with third party applications in the Kubernetes cluster
+- Use Autoscaling to scale your cluster up and down based on utilization
```
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md:28]: lowercase `autoscaling`
or substitute
`the Ray Autoscaler`

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md [hunk 4]
```diff
@@ -2,12 +2,32 @@
...
+- Deploy online Ray applications on Kubernetes
```
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md:29]: Does this refer to the RayService controller?

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md (hunk 1 above)
Comment (DmitriGekhtman) [reply doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md:7]: Fixing my own typo:
`Each Ray cluster consist` -> `Each Ray cluster consists`

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md [hunk 5]
```diff
@@ -2,12 +2,32 @@
...
+    :align: center
+
+..
+  https://docs.google.com/drawings/d/1E3FQgWWLuj8y2zPdKXjoWKrfwgYXw6RV_FWRwK8dVlg/edit
+```
+
+
+Concretely you will learn how to:
+
+- Set up and configure Ray on an existing Kubernetes cluster or a managed Kubernetes service
```
Comment (stephanie-wang) [inline doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md:24]: This list seems a bit long to me and also some of the items seem a bit arbitrary or too specific to me. I wonder if we can organize it better? Here's one cut:

- Set up and configure Ray on a Kubernetes cluster
- Deploy and monitor Ray applications
- Integrate Ray with third-party services

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md (hunk 1 above)
Comment (pcmoritz) [reply doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md]: thanks, fixed

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md (hunk 5 above)
Comment (pcmoritz) [reply doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md]: Thanks, I simplified this now along these lines :)

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md (hunk 4 above)
Comment (pcmoritz) [reply doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md]: This is now removed. I think we should integrate running Serve on KubeRay in the examples but label it as "alpha" -- we can probably start by integrating the material in https://ray-project.github.io/kuberay/guidance/rayservice/ :)

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md (hunk 3 above)
Comment (pcmoritz) [reply doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md]: Removed now according to Stephanie's suggestion -- autoscaling is mentioned in the intro so it should be fine :)

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md (hunk 2 above)
Comment (pcmoritz) [reply doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/index.md]: removed now :)

**Review by DmitriGekhtman — APPROVED** [review]

**Pull request merged**